git-cliff-2.7.0/man/git-cliff.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH git-cliff 1  "git-cliff 2.7.0" 
.SH NAME
git\-cliff \- A highly customizable changelog generator ⛰️
.SH SYNOPSIS
\fBgit\-cliff\fR [\fB\-h\fR|\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] [\fB\-v\fR|\fB\-\-verbose\fR]... [\fB\-i\fR|\fB\-\-init\fR] [\fB\-c\fR|\fB\-\-config\fR] [\fB\-w\fR|\fB\-\-workdir\fR] [\fB\-r\fR|\fB\-\-repository\fR] [\fB\-\-include\-path\fR] [\fB\-\-exclude\-path\fR] [\fB\-\-tag\-pattern\fR] [\fB\-\-with\-commit\fR] [\fB\-\-with\-tag\-message\fR] [\fB\-\-ignore\-tags\fR] [\fB\-\-count\-tags\fR] [\fB\-\-skip\-commit\fR] [\fB\-p\fR|\fB\-\-prepend\fR] [\fB\-o\fR|\fB\-\-output\fR] [\fB\-t\fR|\fB\-\-tag\fR] [\fB\-\-bump\fR] [\fB\-\-bumped\-version\fR] [\fB\-b\fR|\fB\-\-body\fR] [\fB\-l\fR|\fB\-\-latest\fR] [\fB\-\-current\fR] [\fB\-u\fR|\fB\-\-unreleased\fR] [\fB\-\-topo\-order\fR] [\fB\-\-use\-branch\-tags\fR] [\fB\-\-no\-exec\fR] [\fB\-x\fR|\fB\-\-context\fR] [\fB\-\-from\-context\fR] [\fB\-s\fR|\fB\-\-strip\fR] [\fB\-\-sort\fR] [\fB\-\-github\-token\fR] [\fB\-\-github\-repo\fR] [\fB\-\-gitlab\-token\fR] [\fB\-\-gitlab\-repo\fR] [\fB\-\-gitea\-token\fR] [\fB\-\-gitea\-repo\fR] [\fB\-\-bitbucket\-token\fR] [\fB\-\-bitbucket\-repo\fR] [\fIRANGE\fR] 
.SH DESCRIPTION
A highly customizable changelog generator ⛰️
.SH OPTIONS
.TP
\fB\-h\fR, \fB\-\-help\fR
Prints help information
.TP
\fB\-V\fR, \fB\-\-version\fR
Prints version information
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Increases the logging verbosity
.TP
\fB\-i\fR, \fB\-\-init\fR=\fICONFIG\fR
Writes the default configuration file to cliff.toml
.TP
\fB\-c\fR, \fB\-\-config\fR=\fIPATH\fR [default: cliff.toml]
Sets the configuration file
.RS
May also be specified with the \fBGIT_CLIFF_CONFIG\fR environment variable. 
.RE
.TP
\fB\-w\fR, \fB\-\-workdir\fR=\fIPATH\fR
Sets the working directory
.RS
May also be specified with the \fBGIT_CLIFF_WORKDIR\fR environment variable. 
.RE
.TP
\fB\-r\fR, \fB\-\-repository\fR=\fIPATH\fR
Sets the git repository
.RS
May also be specified with the \fBGIT_CLIFF_REPOSITORY\fR environment variable. 
.RE
.TP
\fB\-\-include\-path\fR=\fIPATTERN\fR
Sets the path to include related commits
.RS
May also be specified with the \fBGIT_CLIFF_INCLUDE_PATH\fR environment variable. 
.RE
.TP
\fB\-\-exclude\-path\fR=\fIPATTERN\fR
Sets the path to exclude related commits
.RS
May also be specified with the \fBGIT_CLIFF_EXCLUDE_PATH\fR environment variable. 
.RE
.TP
\fB\-\-tag\-pattern\fR=\fIPATTERN\fR
Sets the regex for matching git tags
.RS
May also be specified with the \fBGIT_CLIFF_TAG_PATTERN\fR environment variable. 
.RE
.TP
\fB\-\-with\-commit\fR=\fIMSG\fR
Sets custom commit messages to include in the changelog
.RS
May also be specified with the \fBGIT_CLIFF_WITH_COMMIT\fR environment variable. 
.RE
.TP
\fB\-\-with\-tag\-message\fR=\fIMSG\fR
Sets custom message for the latest release
.RS
May also be specified with the \fBGIT_CLIFF_WITH_TAG_MESSAGE\fR environment variable. 
.RE
.TP
\fB\-\-ignore\-tags\fR=\fIPATTERN\fR
Sets the tags to ignore in the changelog
.RS
May also be specified with the \fBGIT_CLIFF_IGNORE_TAGS\fR environment variable. 
.RE
.TP
\fB\-\-count\-tags\fR=\fIPATTERN\fR
Sets the tags to count in the changelog
.RS
May also be specified with the \fBGIT_CLIFF_COUNT_TAGS\fR environment variable. 
.RE
.TP
\fB\-\-skip\-commit\fR=\fISHA1\fR
Sets commits that will be skipped in the changelog
.RS
May also be specified with the \fBGIT_CLIFF_SKIP_COMMIT\fR environment variable. 
.RE
.TP
\fB\-p\fR, \fB\-\-prepend\fR=\fIPATH\fR
Prepends entries to the given changelog file
.RS
May also be specified with the \fBGIT_CLIFF_PREPEND\fR environment variable. 
.RE
.TP
\fB\-o\fR, \fB\-\-output\fR=\fIPATH\fR
Writes output to the given file
.RS
May also be specified with the \fBGIT_CLIFF_OUTPUT\fR environment variable. 
.RE
.TP
\fB\-t\fR, \fB\-\-tag\fR=\fITAG\fR
Sets the tag for the latest version
.RS
May also be specified with the \fBGIT_CLIFF_TAG\fR environment variable. 
.RE
.TP
\fB\-\-bump\fR=\fIBUMP\fR
Bumps the version for unreleased changes. Optionally with specified version
.TP
\fB\-\-bumped\-version\fR
Prints bumped version for unreleased changes
.TP
\fB\-b\fR, \fB\-\-body\fR=\fITEMPLATE\fR
Sets the template for the changelog body
.RS
May also be specified with the \fBGIT_CLIFF_TEMPLATE\fR environment variable. 
.RE
.TP
\fB\-l\fR, \fB\-\-latest\fR
Processes the commits starting from the latest tag
.TP
\fB\-\-current\fR
Processes the commits that belong to the current tag
.TP
\fB\-u\fR, \fB\-\-unreleased\fR
Processes the commits that do not belong to a tag
.TP
\fB\-\-topo\-order\fR
Sorts the tags topologically
.TP
\fB\-\-use\-branch\-tags\fR
Include only the tags that belong to the current branch
.TP
\fB\-\-no\-exec\fR
Disables the external command execution
.TP
\fB\-x\fR, \fB\-\-context\fR
Prints changelog context as JSON
.TP
\fB\-\-from\-context\fR=\fIPATH\fR
Generates changelog from a JSON context
.RS
May also be specified with the \fBGIT_CLIFF_CONTEXT\fR environment variable. 
.RE
.TP
\fB\-s\fR, \fB\-\-strip\fR=\fIPART\fR
Strips the given parts from the changelog
.br

.br
[\fIpossible values: \fRheader, footer, all]
.TP
\fB\-\-sort\fR=\fISORT\fR [default: oldest]
Sets sorting of the commits inside sections
.br

.br
[\fIpossible values: \fRoldest, newest]
.TP
\fB\-\-github\-token\fR=\fITOKEN\fR
Sets the GitHub API token
.RS
May also be specified with the \fBGITHUB_TOKEN\fR environment variable. 
.RE
.TP
\fB\-\-github\-repo\fR=\fIOWNER/REPO\fR
Sets the GitHub repository
.RS
May also be specified with the \fBGITHUB_REPO\fR environment variable. 
.RE
.TP
\fB\-\-gitlab\-token\fR=\fITOKEN\fR
Sets the GitLab API token
.RS
May also be specified with the \fBGITLAB_TOKEN\fR environment variable. 
.RE
.TP
\fB\-\-gitlab\-repo\fR=\fIOWNER/REPO\fR
Sets the GitLab repository
.RS
May also be specified with the \fBGITLAB_REPO\fR environment variable. 
.RE
.TP
\fB\-\-gitea\-token\fR=\fITOKEN\fR
Sets the Gitea API token
.RS
May also be specified with the \fBGITEA_TOKEN\fR environment variable. 
.RE
.TP
\fB\-\-gitea\-repo\fR=\fIOWNER/REPO\fR
Sets the Gitea repository
.RS
May also be specified with the \fBGITEA_REPO\fR environment variable. 
.RE
.TP
\fB\-\-bitbucket\-token\fR=\fITOKEN\fR
Sets the Bitbucket API token
.RS
May also be specified with the \fBBITBUCKET_TOKEN\fR environment variable. 
.RE
.TP
\fB\-\-bitbucket\-repo\fR=\fIOWNER/REPO\fR
Sets the Bitbucket repository
.RS
May also be specified with the \fBBITBUCKET_REPO\fR environment variable. 
.RE
.TP
[\fIRANGE\fR]
Sets the commit range to process
.SH VERSION
v2.7.0
.SH AUTHORS
git\-cliff contributors <git\-cliff@protonmail.com>


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## [unreleased]

### 🚀 Features

- *(layout)* Allow reversing the layout and placing input bar on top (#76)

### 🐛 Bug Fixes

- *(output)* Quote output string when it contains spaces and points to an existing path (#77)
- *(stdin)* Better handling of long running stdin streams (#81)

### 🚜 Refactor

- *(exit)* Use std::process::exit explicitly (#84)

### 📚 Documentation

- *(install)* Update the installation section of the README (#79)

### ⚡ Performance

- Remove unnecessary clone() calls (#83)

### ⚙️ Miscellaneous Tasks

- *(changelog)* Update changelog (auto) (#74)

## [0.5.3] - 2024-11-24

### 🚀 Features

- *(navigation)* Add action to scroll results list by a page (#72)

### 🐛 Bug Fixes

- Quote file names that contain spaces when printing them to stdout (#51)
- *(entry)* Always preserve raw input + match ranges conversions (#62)

### 🚜 Refactor

- *(picker)* Refactor picker logic and add tests to picker, cli, and events (#57)

### 📚 Documentation

- Terminal emulators compatibility and good first issues (#56)
- *(contributing)* Added TOC and Code of Conduct link (#65)

### ⚡ Performance

- *(preview)* Cap the number of concurrent preview tokio tasks in the background (#67)

### 🎨 Styling

- *(git)* Enforce conventional commits on git push with a hook (#61)

### ⚙️ Miscellaneous Tasks

- Add readme version update to github actions (#55)
- *(update_readme)* Fix `update_readme` workflow (#63)
- *(changelog)* Update changelog action trigger (#68)
- *(changelog)* Update changelog (auto) (#70)
- *(changelog)* Update changelog (auto) (#73)
- Bump crate to 0.5.3 and workspace crates to 0.0.7

### Build

- *(infer)* Drop infer dependency and refactor code to a simpler heuristic (#58)

## [0.5.1] - 2024-11-20

### 📚 Documentation

- Add instructions for installing on Arch Linux (#43)
- *(brew)* Add brew installation method for MacOS to README (#45)
- *(config)* Update docs to mention XDG_CONFIG_HOME precedence on all platform (#48)

### ⚙️ Miscellaneous Tasks

- Add CHANGELOG.md (#44)
- *(config)* Default configuration now uses 100% of terminal screen space (#47)
- *(changelog)* Udpate changelog and add corresponding makefile command (#53)
- *(actions)* Remove changelog update from the main branch
- Bump version to 0.5.1

## [0.5.0] - 2024-11-18

### 🚀 Features

- *(cli)* Allow passing passthrough keybindings via stdout for the parent process to deal with (#39)
- *(ui)* Make the top UI help bar toggleable (#41)

### 🚜 Refactor

- *(config)* [**breaking**] Make action names snake case in keybinding configuration (#40)

### 📚 Documentation

- Update README television version
- Update README television version specifier
- Update README television version

### ⚙️ Miscellaneous Tasks

- *(rustfmt)* Update rustfmt.toml (#42)

## [0.4.23] - 2024-11-16

### 🚀 Features

- *(ui)* Make help bar display optional (#35)

### 🚜 Refactor

- *(configuration)* Modularize code and better handling of default options (#32)

### 📚 Documentation

- *(debian)* Add installation docs for debian-based systems (#33)
- *(config)* Update docs default configuration (#34)

## [0.4.22] - 2024-11-16

### 🐛 Bug Fixes

- *(ghactions)* Only trigger cd workflow on new tags (#22)
- *(config)* Swap out default keymaps with user defined ones instead of stacking (#26)

### 🚜 Refactor

- *(channels)* Converting between entries and channels is now generic over channels (#25)

### ⚙️ Miscellaneous Tasks

- *(deb)* Release deb package for television (#31)
- Update CD workflow
- *(cd)* Fix cd configuration for deb packages
- *(cd)* Fix cd configuration for deb packages
- *(versions)* Bump workspace crates versions

## [0.4.21] - 2024-11-13

### 🐛 Bug Fixes

- *(windows)* #20 respect `TELEVISION_CONFIG` env var on windows (#21)

### ⚙️ Miscellaneous Tasks

- *(nix)* Nix flake shell + rust-toolchain.toml setup (#14)

## [0.4.20] - 2024-11-11

### 🐛 Bug Fixes

- *(windows)* Ignore `KeyEventKind::Release` events (#3)
- *(windows)* Bump television_utils to v0.0.1 (#4)
- *(config)* More consistent configuration file location for linux and macos (#9)
- *(workspace)* Fix cargo workspace dependencies
- *(cargo workspace)* Fix cargo workspace structure and dependencies (#15)

### 🚜 Refactor

- *(workspace)* Reorganize cargo workspace (#12)

### 📚 Documentation

- Add terminal emulators compatibility status
- Fix table alignments
- *(readme)* Update terminal emulators compatibility list (#6)

### ⚡ Performance

- *(preview)* Remove temporary plaintext previews in favor of loading message preview (#10)

### ⚙️ Miscellaneous Tasks

- Update README.md install section
- *(coc)* Create CODE_OF_CONDUCT.md (#7)
- *(crate)* Add include directives to Cargo.toml to make the crate leaner (#11)

## [0.4.18] - 2024-11-10

### 🐛 Bug Fixes

- Add the correct permissions to release binaries
- Add `winapi-util` dependency for windows builds

## [0.4.17] - 2024-11-10

### ⚙️ Miscellaneous Tasks

- Udate documentation and dependencies
- Update Makefile and CONTRIBUTING.md
- Testing out the CD pipeline

## [0.4.15] - 2024-11-10

### 🚀 Features

- Send to channel
- More syntaxes and themes for highlighting + configuration

### 🐛 Bug Fixes

- Fixing various issues
- Fixing various issues
- Fix linting issues and ignore derive docs for tests
- Filtering system directories in gitrepos
- Stabilize preview scroll initialization
- Doctests imports
- Gag stdout and stderr while loading theme assets to silence bat warning

### 🚜 Refactor

- Refactoring
- Refactoring matcher
- Extract matcher logic into separate crate
- Split project into separate crates
- More refactoring and fixing doctests

### 📚 Documentation

- Docs and linting
- Documentation
- Update README.md
- Add default keybindings to README.md
- Some work on CONTRIBUTING.md
- More work on CONTRIBUTING.md

### 🧪 Testing

- Tests and docs for strings.rs
- Testing ci

### ⚙️ Miscellaneous Tasks

- Bump version
- Bump version

git-cliff-2.7.0/completions/git-cliff.bash
_git-cliff() {
    local i cur prev opts cmd
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="git__cliff"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        git__cliff)
            opts="-h -V -v -i -c -w -r -p -o -t -b -l -u -x -s --help --version --verbose --init --config --workdir --repository --include-path --exclude-path --tag-pattern --with-commit --with-tag-message --ignore-tags --count-tags --skip-commit --prepend --output --tag --bump --bumped-version --body --latest --current --unreleased --topo-order --use-branch-tags --no-exec --context --from-context --strip --sort --github-token --github-repo --gitlab-token --gitlab-repo --gitea-token --gitea-repo --bitbucket-token --bitbucket-repo [RANGE]"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --init)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -i)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --config)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -c)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --workdir)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -w)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --repository)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -r)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --include-path)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --exclude-path)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --tag-pattern)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --with-commit)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --with-tag-message)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --ignore-tags)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --count-tags)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --skip-commit)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --prepend)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -p)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --output)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --tag)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -t)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --bump)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --body)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -b)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --from-context)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --strip)
                    COMPREPLY=($(compgen -W "header footer all" -- "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -W "header footer all" -- "${cur}"))
                    return 0
                    ;;
                --sort)
                    COMPREPLY=($(compgen -W "oldest newest" -- "${cur}"))
                    return 0
                    ;;
                --github-token)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --github-repo)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --gitlab-token)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --gitlab-repo)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --gitea-token)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --gitea-repo)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --bitbucket-token)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --bitbucket-repo)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

if [[ "${BASH_VERSINFO[0]}" -eq 4 && "${BASH_VERSINFO[1]}" -ge 4 || "${BASH_VERSINFO[0]}" -gt 4 ]]; then
    complete -F _git-cliff -o nosort -o bashdefault -o default git-cliff
else
    complete -F _git-cliff -o bashdefault -o default git-cliff
fi
